### PR TITLE
some care for unused variables in matrix/

### PR DESCRIPTION
**Pull request by fchapoton**
partial care for unused variables in matrix folder, following suggestions of cython-lint

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

**Comment by fchapoton** [comment]
build failure probably unrelated

File: src/sage/matrix/matrix_double_dense.pyx [hunk 1]
```diff
@@ -1739,14 +1739,13 @@ cdef class Matrix_double_dense(Matrix_numpy_dense):
             import scipy
         import scipy.linalg
         X = self._new(self._ncols, B.ncols())
-        arr, resid, rank, s = scipy.linalg.lstsq(self._matrix_numpy, B.numpy())
+        arr, _, _, _ = scipy.linalg.lstsq(self._matrix_numpy, B.numpy())
```
Comment (dcoudert) [inline src/sage/matrix/matrix_double_dense.pyx]: do you know this funny Python syntax ? could help avoiding `_, _, _`
```
sage: a, *b = list(range(5))
sage: a, b
(0, [1, 2, 3, 4])
sage: *a, b = list(range(5))
sage: a, b
([0, 1, 2, 3], 4)
sage: a, *_, b = list(range(50))
sage: a, b
(0, 49)
```

Comment (fchapoton) [reply src/sage/matrix/matrix_double_dense.pyx]: merci. J'ai choisi de faire un truc plus simple.

**Review by dcoudert — APPROVED** [review]
LGTM. It passes tests on my macOS laptop.

May be you could fix this doctest warning in this ticket and then set it to positive review on my behalf.
```
sage -t --long --warn-long 32.4 --random-seed=206992355279890304379498852466385141948 src/sage/matrix/matrix1.pyx
**********************************************************************
File "src/sage/matrix/matrix1.pyx", line 729, in sage.matrix.matrix1.Matrix.numpy
Warning: Variable 'b' referenced here was set only in doctest marked '# needs numpy'
    b.shape
    [449 tests, 1.81 s]
```

**Comment by fchapoton** [comment]
not sure I want to touch that. Really not. There is a "needs numpy" for the full block, and it is the "32 bit / 64 bit tags" that breaks this awful tagging stuff. Not my business, really.

**Comment by dcoudert** [comment]
Right, there is an open issue for that.

**Comment by fchapoton** [comment]
merci David

**Pull request merged**